frontend/src/pages/LogsPage/LogsPage.tsx

```diff
@@ -307,7 +303,11 @@ const LogsPageInner = ({ timeMode, logCursor, presetDefault }: Props) => {
 						timeMode={timeMode}
 						savedSegmentType={SavedSegmentEntityType.Log}
 						textAreaRef={textAreaRef}
-						enableAIMode={enableAiQueryBuilder}
+						enableAIMode={
+							workspaceSettings?.workspaceSettings
+								?.ai_query_builder
+						}
+						aiSupportedSearch={aiQueryBuilderFlag}
```

Comment:
2 options:
1. Search supports the AI query build
2. The query builder is enabled in the user's workspace